### PR TITLE
Version 1.2.3

### PR DESCRIPTION
**Pull request by nateshim-indico**
- Upgrades PyMuPDF to v1.19.6 and reverts it as non-optional dependency
- Removes warnings generated in test suite

### CODE REVIEW

File: tests/pytest.ini [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[pytest]
```
Comment (nateshim-indico) [inline tests/pytest.ini:1]: Created this to resolve PytestUnknownMarkWarning being generated: https://stackoverflow.com/questions/60806473/pytestunknownmarkwarning-unknown-pytest-mark-xxx-is-this-a-typo

File: indico_toolkit/snapshots/snapshot.py [hunk 2]
```diff
@@ -93,7 +93,7 @@ def append(self, snap_to_add: Snapshot):
             snap_to_add (Snapshot): Snapshot to add
         """
         self._assert_key_column_names_match(snap_to_add)
-        self.df = self.df.append(snap_to_add.df, ignore_index=True)
+        self.df = pd.concat([self.df, snap_to_add.df], ignore_index=True)
```
Comment (nateshim-indico) [inline indico_toolkit/snapshots/snapshot.py:96]: Apparently, pandas append is getting deprecated: https://stackoverflow.com/questions/70837397/good-alternative-to-pandas-append-method-now-that-it-is-being-deprecated

File: tests/snapshots/test_snapshot.py [hunk 3]
```diff
@@ -3,7 +3,6 @@
 import tempfile
 from copy import deepcopy
 import pandas as pd
-from pandas.util.testing import assert_frame_equal
```
Comment (nateshim-indico) [inline tests/snapshots/test_snapshot.py:6]: From what I could tell, this was not being used and it apparently is getting deprecated as well: https://github.com/pydata/pandas-datareader/issues/775

File: tests/pytest.ini (hunk 1 above)
Comment (Scott771) [reply tests/pytest.ini:1]: sounds good

File: indico_toolkit/snapshots/snapshot.py (hunk 2 above)
Comment (Scott771) [reply indico_toolkit/snapshots/snapshot.py:96]: makes sense

File: tests/snapshots/test_snapshot.py (hunk 3 above)
Comment (Scott771) [reply tests/snapshots/test_snapshot.py:6]: yup

**Review by Scott771 — APPROVED** [review]

**Pull request merged**